internal/core/saptune/saptune.go

```diff
@@ -0,0 +1,120 @@
+	Schema      string    `json:"$schema"`
+	PublishTime string    `json:"publish time"`
+	Argv        string    `json:"argv"`
+	Pid         int       `json:"pid"`
+	Command     string    `json:"command"`
+	ExitCode    int       `json:"exit code"`
+	Result      Result    `json:"result"`
+	Messages    []Message `json:"messages"`
+}
+
+type Result struct {
+	Services                 Services `json:"services"`
+	SystemdSystemState       string   `json:"systemd system state"`
+	TuningState              string   `json:"tuning state"`
+	Virtualization           string   `json:"virtualization"`
+	ConfiguredVersion        string   `json:"configured version"`
+	PackageVersion           string   `json:"package version"`
+	SolutionEnabled          []string `json:"Solution enabled"`
+	NotesEnabledBySolution   []string `json:"Notes enabled by Solution"`
+	SolutionApplied          []string `json:"Solution applied"`
+	NotesAppliedBySolution   []string `json:"Notes applied by Solution"`
+	NotesEnabledAdditionally []string `json:"Notes enabled additionally"`
+	NotesEnabled             []string `json:"Notes enabled"`
+	NotesApplied             []string `json:"Notes applied"`
+	Staging                  Staging  `json:"staging"`
+	RememberMessage          string   `json:"remember message"`
+}
+
+type Services struct {
+	Saptune []string `json:"saptune"`
+	Sapconf []string `json:"sapconf"`
+	Tuned   []string `json:"tuned"`
+}
+
+type Staging struct {
+	StagingEnabled  bool     `json:"staging enabled"`
+	NotesStaged     []string `json:"Notes staged"`
+	SolutionsStaged []string `json:"Solutions staged"`
+}
+
+type Message struct {
+	Priority string `json:"priority"`
+	Message  string `json:"message"`
+}
+
+func getSaptuneVersion(commandExecutor utils.CommandExecutor) (string, error) {
+	log.Info("Requesting Saptune version...")
+	versionOutput, err := commandExecutor.Exec("rpm", "-q", "--qf", "%{VERSION}", "saptune")
+	if err != nil {
+		return "", ErrSaptuneVersionUnknown
+	}
+
+	log.Debugf("saptune version output: %s", string(versionOutput))
+
+	return string(versionOutput), nil
+}
+
+func isSaptuneVersionSupported(version string) bool {
+
+	compareOutput := semver.Compare(MinimalSaptuneVersion, "v"+version)
+
+	return compareOutput != 1
+}
+
+func NewSaptune(commandExecutor utils.CommandExecutor) (Saptune, error) {
+	saptuneVersion, err := getSaptuneVersion(commandExecutor)
+	if err != nil {
+		return Saptune{Version: nil, Commands: []Output{}}, err
+	}
+
+	if !isSaptuneVersionSupported(saptuneVersion) {
+		return Saptune{Version: &saptuneVersion, Commands: []Output{}}, ErrUnsupportedSaptuneVer
+	}
+
+	log.Info("Requesting Saptune status...")
+	output, err := commandExecutor.Exec("saptune", "--format", "json", "status")
+	if err != nil {
+		return Saptune{Version: &saptuneVersion, Commands: []Output{}},
+			errors.Wrap(err, "unexpected error while calling saptune")
+	}
+
+	log.Debugf("saptune output: %s", string(output))
+
+	var saptuneOutput Output
```

Comment:
Do we need to unmarshall, why don't we send directly the json output in the discovery payload?